src/Hcal/HcalDigiProducer.cxx

```diff
@@ -49,6 +53,52 @@ void HcalDigiProducer::configure(framework::config::Parameters& ps) {
   //  time [ns] * ( 2^10 / max time in ns ) = clock counts
   ns_ = 1024. / clockCycle_;
 
+  // Configure photon generator
+  scintillationYield_= ps.getParameter<double>("scintillationYield");
+  scintillationYieldSigma_= ps.getParameter<double>("scintillationYieldSigma");
+  scintillationYieldCutoffLow_= ps.getParameter<double>("scintillationYieldCutoffLow");
+  scintillationYieldCutoffHigh_= ps.getParameter<double>("scintillationYieldCutoffHigh");
+  std::vector<std::string> lookupTableNames  = ps.getParameter<std::vector<std::string>>("lookupTables");
+
+  // load lookup table for each scintillator length
+  for(size_t i=0; i<lookupTableNames.size(); ++i)
+  {
+    std::shared_ptr<HcalPhotonGenerator> photonGenerator = std::shared_ptr<HcalPhotonGenerator>(new HcalPhotonGenerator(randFlat_, randGaussQ_, randPoissonQ_));
+    photonGenerator->LoadLookupTable(lookupTableNames.at(i),1);
+    photonGenerator->SetScintillationYield(scintillationYield_);
+    int length=(int)(photonGenerator->GetScintillatorLength()+0.5); //rounded to closest int to avoid mismatches due to precission of floating point numbers
+    int reflectorType=photonGenerator->GetReflectorType();
+    photonGenerators_[std::pair<int,int>(length,reflectorType)]=photonGenerator;
+  }
+  scintillationYieldsAdjusted_.clear();
+
+  // Configure charge generator
```

Comment:
@ehrlich-uva do you have a reference for the parameters you chose below? What does pixel mean in the context of the charge generator? do all SiPMs have 40x40?

Reply:
I got them from the specs of the Mu2e CRV SiPMs (Hamamatsu S13360-2050VE). I used these value, because I didn't know the values of the HCal SiPMs. I think it's Ok for now, but should eventually be changed.

Reply:
The SiPM consist of a number of pixels, and each pixel is treated separately (except for cross-talk, where pixels can cause an avalanche at neighboring pixels). The detection probability, dark counts, charging/discharging behavior, avalanche probability and pulse strength are simulated for each pixel individually. Saturation effects are simulated indirectly, too.